Console/UI/Pages/EventsLog.xaml.cs

```diff
@@ -126,94 +140,114 @@ public EventsLog()
         {
+            //LogHelper.Debug($"EntryWritten event ...");
             if (!FirewallHelper.IsEventAccepted(e.Entry))
             {
                 return;
             }
 
-            LogEntryViewModel entry = EntryViewFromLogEntry(e.Entry);
-            if (entry != null)
+            lock (EntriesLock)
             {
-                Dispatcher.Invoke(() =>
+                if (e.Entry != null && e.Entry.Index > SecurityLogEntryIndex_last)
                 {
-                    if (!TargetCollectionNewEntries.Contains(entry, EntryComparer.Instance))
+                    LogEntryViewModel entry = EntryViewFromLogEntry(e.Entry);
+                    if (entry != null)
                     {
-                        TargetCollectionNewEntries.Add(entry);
+                        Dispatcher.Invoke(() =>
+                        {
+                            Dispatcher.Invoke(() => { LogEntries.Add(entry); ScanProgress = LogEntries.Count; });
+                        });
                     }
-                });
+                }
+                if (ScanProgressMax < LogEntries.Count)
+                {
+                    // overflow handling
+                    LogEntries.RemoveAt(0);
+                }
+                SecurityLogEntryIndex_last = securityLog.Entries.Count - 1;
             }
         }
 
-        private static bool TCPFilter(object le) => ((LogEntryViewModel)le).Protocol == "TCP";
-
-        private void InitEventLog(Action<int> progressCallback)
+        /*
+         * Initializes the LogEntries list to show the most recent events
+         */
+        private void InitEventLog(int lastIndex, Action<int> progressCallback)
         {
+            int i = 0;
             try
             {
-                Dispatcher.Invoke(() => LogEntries.Clear());
-
-                ScanProgressMax = securityLog.Entries.Count;
-
-                // Have to use a Enumerable.Range to enumerate the entries starting from the last one, since calling Reverse on the initial collection will first enumerate all entries
-                var entries = Enumerable.Range(0, ScanProgressMax)
-                                        .Select(i => { progressCallback?.Invoke(i + 1); return securityLog.Entries[ScanProgressMax - i - 1]; })
-                                        .Where(FirewallHelper.IsEventAccepted)
-                                        .Select(EntryViewFromLogEntry)
-                                        .Where(entry => entry != null)
-                                        .Select(entry => { Dispatcher.Invoke(() => { LogEntries.Add(entry); }); return entry; })
-                                        .ToList();
+                LogHelper.Debug($"InitEventLog ...");
+                for (i = lastIndex; i >= 0; i--)
+                {
+                    lock (EntriesLock)
+                    {
+                        if (i > securityLog.Entries.Count - 1) { continue; }

```

Comment:
I case this is in check the latest entry was in fact not yet added to the list?

Reply:
Well, it's probably not necessary anmore - the out of bounds exceptions occured all over the place 🤔

Reply:
btw: It's possible that "securityLog.Entries.Count" changes as well, because the securityLog limits the number of entries  (somehow)  - from what I've noticed, it stays around 30K rows on my system i.e. it doesn't grow forever.